oprf/oprf.go

```diff
@@ -0,0 +1,238 @@
+package oprf
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+
+	"github.com/cloudflare/circl/oprf/group"
+)
+
+const (
+	// OPRFCurve25519 is the constant to represent the OPRF curve25519 with SHA-512 (ELL2-RO) group.
+	OPRFCurve25519 uint16 = 0x0001
+	// OPRFCurve448 is the constant to represent the OPRF curve448 with SHA-512 (ELL2-RO) group.
+	OPRFCurve448 uint16 = 0x0002
+	// OPRFP256 is the constant to represent the OPRF P-256 with SHA-512 (SSWU-RO) group.
+	OPRFP256 uint16 = 0x0003
+	// OPRFP384 is the constant to represent the OPRF P-384 with SHA-512 (SSWU-RO) group.
+	OPRFP384 uint16 = 0x0004
+	// OPRFP521 is the constant to represent the OPRF P-521 with SHA-512 (SSWU-RO) group.
+	OPRFP521 uint16 = 0x0005
+)
+
+const (
+	// OPRFMode is the context string to define a OPRF.
+	OPRFMode string = "000"
+)
+
+var (
+	// ErrUnsupportedGroup is an error stating that the ciphersuite chosen is not supported
+	ErrUnsupportedGroup = errors.New("the chosen group is not supported")
+)
+
+// BlindToken corresponds to a token that has been blinded.
+// Internally, it is a serialized Element.
+type BlindToken []byte
+
+// IssuedToken corresponds to a token that has been issued.
+// Internally, it is a serialized Element.
+type IssuedToken []byte
+
+// Token is the object issuance of the protocol.
+type Token struct {
+	data  []byte
+	blind *group.Scalar
+}
+
+// Evaluation corresponds to the evaluation over a token.
+type Evaluation struct {
+	element []byte
+}
+
+// KeyPair is an struct containing a public and private key.
+type KeyPair struct {
+	PubK  *group.Element
```

Comment:
Do these need to be exported?

Reply:
I'm not sure. Do the OPAQUE or HIBPvN libraries will need them?